### DISCUSSION TITLE
GPT-2-LM-Head ONNX file input

### DISCUSSION THREAD

**Discussion by cayleyhamilton** [microsoft/onnxruntime#7370] [Other Q&A]
https://github.com/onnx/models/tree/master/text/machine_comprehension/gpt-2
The ONNX file's input is int64[input1_dyanamic_axes_1, input1_dynamic_axes_2, input1_dynamic_axes_3]

The referring text says 'long tensor of dynamic shape (batch_size, sequence_length).  This makes intuitive sense to me. 
A tensor with a token-id for each batch element and word piece.

How do I interpret the 3 axes in the ONNX input? 
How is the variable sequence length within a batch represented?

Thanks for your help, in advance.

**Comment by pranavsharma** [comment]
I suspect the shape of the model's input is incorrect. It should indeed be (batch_size, sequence_length). cc @neginraoof - can you please check? Also, this issue is better filed here https://github.com/onnx/models.
